File: app/src/main/java/br/com/santanderteste/ui/fragment/LoginFragment.java
package br.com.santanderteste.ui.fragment;

import android.arch.persistence.room.Room;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.navigation.NavOptions;
import androidx.navigation.Navigation;
import br.com.santanderteste.R;
import br.com.santanderteste.presenter.LoginPresenter;
import br.com.santanderteste.repository.UserRepository;
import br.com.santanderteste.repository.db.UserDatabase;
import br.com.santanderteste.ui.BaseFragment;
import br.com.santanderteste.ui.activity.MainActivity;
import br.com.santanderteste.ui.interfaces.ILoginView;
import br.com.santanderteste.utils.Const;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

/**
 * @author JhonnyBarbosa
 * @version 1.0
 */
public class LoginFragment extends BaseFragment implements ILoginView {

    private LoginPresenter loginPresenter;
    private UserRepository userRepository;

    private Unbinder unbinder;
    private UserDatabase userDatabase;

    @BindView(R.id.login_button)
    Button loginButton;

    @BindView(R.id.username_login)
    EditText userNameLogin;

    @BindView(R.id.password)
    EditText password;

    @BindView(R.id.login_progressbar)
    ProgressBar loginProgressBar;

    @BindView(R.id.error_message)
    TextView errorMessage;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.login_layout, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        hideActionBar((MainActivity) getActivity());

        userDatabase = Room.databaseBuilder(getActivity(),
                UserDatabase.class, Const.DATABASE_NAME)
                .fallbackToDestructiveMigration()
                .build();

        userRepository = new UserRepository(userDatabase);
        loginPresenter = new LoginPresenter(this, userRepository);

        userNameLogin.setText("51544156049");
        password.setText("Test@1");

        loginPresenter.loadUserData();

    }

    @Override
    public boolean isViewAdde() {
        return isAdded();
    }

    @Override
    public void callStatementsFragment(Bundle bundle) {
        NavOptions options = new NavOptions.Builder().setPopUpTo(R.id.loginFragment, true).build();
        Navigation.findNavController(getActivity(), R.id.main_nav_host).navigate(R.id.StatementFragment, bundle, options);
    }

    @Override
    public void showProgress() {
        loginProgressBar.setVisibility(View.VISIBLE);
        loginButton.setText(Const.EMPTY_STRING);
    }

    @Override
    public void hideProgress() {
        loginProgressBar.setVisibility(View.GONE);
        loginButton.setText(Const.LOGIN);
    }

    @Override
    public void disableEditText() {
        userNameLogin.setEnabled(false);
        password.setEnabled(false);
    }


    @Override
    public void enableEditText() {
        userNameLogin.setEnabled(true);
        password.setEnabled(true);
    }

    @Override
    public void showErrorMessage(String message) {
        errorMessage.setText(message);
        errorMessage.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideErrorMessage() {
        errorMessage.setVisibility(View.GONE);
    }

    @OnClick(R.id.login_button)
    public void onClickLogin(View view) {

        String u = userNameLogin.getText().toString();
        String p = password.getText().toString();

        loginPresenter.checkLoginData(u, p);

    }

}

File: Bank/app/src/main/java/com/appdesafioSantander/bank/ui/login/LoginFragment.kt
package com.appdesafioSantander.bank.ui.login

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.findNavController
import com.appdesafioSantander.bank.R
import com.appdesafioSantander.bank.model.Login
import com.appdesafioSantander.bank.model.LoginViewModel
import com.appdesafioSantander.bank.utils.ViewUtils

class LoginFragment : Fragment(), LoginListener {

    lateinit var loginViewModel: LoginViewModel
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        loginViewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
        loginViewModel.loginListener = this
        return inflater.inflate(R.layout.login, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val loginBtn = view.findViewById<Button>(R.id.login_btn)
        loginBtn.setOnClickListener {
            loginViewModel.onLoginButtonClick(view)
        }
    }

    override fun onStarted(view: View) {
        super.onStarted(view)
        ViewUtils.show(view)
    }

    override fun onSuccess(view: View, login: LiveData<Login>) {
        super.onSuccess(view, login)
        login.observe(this, Observer {
            if(it.error.isNullOrEmpty()){
                val bundle = bundleOf("user" to it.userAccount)
                view.findNavController().navigate(R.id.action_Login_to_Statements, bundle)
            } else {
                ViewUtils.toast(activity!!.baseContext, "${it.error["message"]}")
            }
        })
    }

    override fun onFailure(view: View, msg: String) {
        super.onFailure(view, msg)
        ViewUtils.hide(view)
        ViewUtils.toast(activity!!.baseContext, msg)
    }
}